v3.24.0 -> panic: interface conversion: interface {} is *schema.Set, not []interface {}

PagerDuty v3.24.0

╷
│ Error: Request cancelled
│ 
│   with module.pagerduty_service.pagerduty_service_integration.datadog,
│   on ../modules/pagerduty-service/service.tf line 28, in resource "pagerduty_service_integration" "datadog":
│   28: resource "pagerduty_service_integration" "datadog" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.pagerduty_service.pagerduty_slack_connection.team[0],
│   on ../modules/pagerduty-service/slack.tf line 19, in resource "pagerduty_slack_connection" "team":
│   19: resource "pagerduty_slack_connection" "team" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.pagerduty_service.pagerduty_slack_connection.stakeholder[0],
│   on ../modules/pagerduty-service/slack.tf line 39, in resource "pagerduty_slack_connection" "stakeholder":
│   39: resource "pagerduty_slack_connection" "stakeholder" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-pagerduty_v3.24.0 plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 132 [running]:
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.expandConnectionConfig({0x1064fa0?, 0xc0007ae8a0?})
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_slack_connection.go:224 +0x4c5
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.buildSlackConnectionStruct(0xc0004c5c00)
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_slack_connection.go:110 +0x225
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.resourcePagerDutySlackConnectionCreate.func1()
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_slack_connection.go:122 +0x31
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.RetryContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/wait.go:30 +0x4e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:113 +0x1a3
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext in goroutine 102
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:86 +0x237

Error: The terraform-provider-pagerduty_v3.24.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/pagerduty/resource_pagerduty_slack_connection.go
+++ b/pagerduty/resource_pagerduty_slack_connection.go
@@ -221,7 +221,7 @@ func expandConnectionConfig(v interface{}) pagerduty.ConnectionConfig {
 	var config pagerduty.ConnectionConfig
 
 	config = pagerduty.ConnectionConfig{
-		Events:     expandConfigList(c["events"].([]interface{})),
+		Events:     expandConfigList(c["events"].(*schema.Set).List()),
 		Priorities: expandStarWildcardConfig(expandConfigList(c["priorities"].([]interface{}))),
 		Urgency:    nil,
 	}

